firefox-ios/Client/AccountSyncHandler.swift

```diff
@@ -11,10 +11,13 @@ import Common
 class AccountSyncHandler: TabEventHandler {
```

Comment:
Note: this handler should probably remain "global" in the sense that we should not create separate instances for each window, since the user's `Profile` is effectively part of the app's global state. There are some aspects of this tab selection syncing which I need to revisit, so I'm filing a separate ticket to follow-up on this. For now I've kept this handler following the same pattern used elsewhere in this PR (which makes it window-specific), but that will be revisited soon. For now (in current `main`) this should impart no actual behavioral changes.